r/tests/testthat/test-dplyr.R

```diff
@@ -501,6 +503,7 @@ test_that("explicit type conversions with as.*()", {
 })
 
 test_that("as.factor()/dictionary_encode()", {
+  skip("ExecuteScalarExpression cannot Execute non-scalar expression {x=dictionary_encode(x, {NON-REPRESENTABLE OPTIONS})}")
```

Comment:
Adding support for dictionary encoding is possible but it will be subtle since it will be the first Expression with mutable state to share between threads. I've added https://issues.apache.org/jira/browse/ARROW-12632 to track this